docs/content/concepts/assets/asset-checks.mdx

```diff
@@ -0,0 +1,153 @@
+---
+title: Asset checks | Dagster
+description: Asset checks are a way to define expectations about the quality of your data.
+---
+
+# Asset checks (Experimental)
+
+Dagster allows you to define and execute data quality checks on your [software-defined assets](/concepts/assets/software-defined-assets). Each asset check verifies some property of a data asset, e.g. that is has no null values in a particular column.
+
+When viewing an asset in Dagster’s UI, you can see all of its checks, and whether they’ve passed, failed, or haven’t run.
+
+## Defining asset checks
+
+### Asset check that executes in its own op
+
+The following code defines an asset named `orders` and an asset check named `orders_id_is_unique`. When executed, the check verifies a property of the `orders` asset: that all the values in its primary key column are unique.
+
+```python file=/concepts/assets/asset_checks/orders_check.py
+import pandas as pd
+
+from dagster import AssetCheckResult, Definitions, asset, asset_check
+
+
+@asset
+def orders():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    orders_df.to_csv("orders.csv")
+
+
+@asset_check(asset=orders, description="Check for null order_ids")
+def orders_id_has_no_nulls():
+    orders_df = pd.read_csv("orders.csv")
+    num_null_order_ids = orders_df["order_id"].isna().sum()
+    return AssetCheckResult(
+        success=bool(num_null_order_ids == 0),
```

Comment:
does this need to be on its own line? also, why does it need to be cast to a bool? (might have been my fault in the first place)

Reply:
it's a numpy bool. Good motivation for something like https://dagsterlabs.slack.com/archives/C05KFG9TETB/p1694729820023879